pkg/sql/internal.go

```diff
@@ -72,6 +78,11 @@ type InternalExecutor struct {
 	//
 	// Warning: Not safe for concurrent use from multiple goroutines.
 	syntheticDescriptors []catalog.Descriptor
+
+	// extraTxnState is to store extra transaction state info that
+	// will be passed to an internal executor. It should only be set when the
+	// internal executor is used under a not-nil txn.
+	extraTxnState *extraTxnState
```

Comment:
can you put a TODO that this is a wart that we never really wanted to exist in the first place and intend to remove ASAP by tightly coupling the internal executor to a connExecutor instance? I'm afraid that if we just leave this here some future person might think it was something we intended to invest in when in fact it's something we detest.

Reply:
Done